### ISSUE TITLE
[CI] Netty4HttpRequestSizeLimitIT#testDoesNotLimitExcludedRequests failure

### ISSUE THREAD

**Issue by martijnvg** [elastic/elasticsearch#29217]
Does not reproduce locally. Initally I thought it maybe a failure that reproduce with java10, but after running this test with Java 10 I was not able to make this test fail.

```
  2> REPRODUCE WITH: ./gradlew :modules:transport-netty4:integTestRunner -Dtests.seed=BD62D596B9434C89 -Dtests.class=org.elasticsearch.http.netty4.Netty4HttpRequestSizeLimitIT -Dtests.method="testDoesNotLimitExcludedRequests" -Dtests.security.manager=true -Dtests.locale=en-ZA -Dtests.timezone=Asia/Barnaul
  1> [2018-03-23T14:16:26,879][INFO ][o.e.n.Node               ] [node_t0] stopping ...
  1> [2018-03-23T14:16:26,889][INFO ][o.e.n.Node               ] [node_t0] stopped
  1> [2018-03-23T14:16:26,889][INFO ][o.e.n.Node               ] [node_t0] closing ...
  1> [2018-03-23T14:16:26,900][INFO ][o.e.n.Node               ] [node_t0] closed
  1> [2018-03-23T14:16:26,900][INFO ][o.e.h.n.Netty4HttpRequestSizeLimitIT] [Netty4HttpRequestSizeLimitIT#testDoesNotLimitExcludedRequests]: cleaned up after test
  1> [2018-03-23T14:16:26,900][INFO ][o.e.h.n.Netty4HttpRequestSizeLimitIT] [testDoesNotLimitExcludedRequests]: after test
FAILURE 14.9s | Netty4HttpRequestSizeLimitIT.testDoesNotLimitExcludedRequests <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: a collection with size <1500>
   >      but: collection size was <1489>
   > 	at __randomizedtesting.SeedInfo.seed([BD62D596B9434C89:47136C0FB4BD16C9]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.http.netty4.Netty4HttpRequestSizeLimitIT.testDoesNotLimitExcludedRequests(Netty4HttpRequestSizeLimitIT.java:113)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   > 	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
   > 	at java.base/java.lang.Thread.run(Thread.java:844)
  1> [2018-03-23T14:16:26,909][INFO ][o.e.h.n.Netty4HttpRequestSizeLimitIT] [testLimitsInFlightRequests]: before test
  1> [2018-03-23T14:16:26,909][INFO ][o.e.h.n.Netty4HttpRequestSizeLimitIT] [Netty4HttpRequestSizeLimitIT#testLimitsInFlightRequests]: setting up test
  1> [2018-03-23T14:16:26,910][INFO ][o.e.t.InternalTestCluster] Setup InternalTestCluster [TEST-CHILD_VM=[0]-CLUSTER_SEED=[-4361463119098830266]-HASH=[D61B8210D83B3]-cluster] with seed [C378F866B525A246] using [0] dedicated masters, [1] (data) nodes and [0] coord only nodes (min_master_nodes are [auto-managed])
```

[consoleText.txt](https://github.com/elastic/elasticsearch/files/1840684/consoleText.txt)

Build url: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.2+multijob-java-periodic/ESJAVA=java9,ESRUNTIME=openjdk10,nodes=linux/18/console

**Comment by jasontedor** [comment]
This is a race condition in the test. Namely, we set a latch for 10 seconds for all the requests to come back by. Here, we can see that that 10 seconds expired before all the requests returned:

```
  1> [2018-03-23T14:16:14,706][INFO ][o.e.c.s.ClusterSettings  ] [node_t0] updating [search.default_search_timeout] from [-1] to [40s]
  1> [2018-03-23T14:16:26,848][INFO ][o.e.h.n.Netty4HttpRequestSizeLimitIT] [Netty4HttpRequestSizeLimitIT#testDoesNotLimitExcludedRequests]: cleaning up after test
```

I will increase the time bound on the latch. If we continue to see this as an issue (I looked and did not see it occurring much, 10 seconds is a long time and really is only going to trip if the CI worker is under load or guest contention) I will consider another option (such as removing the latch and just letting the test timeout with our default timeout).
